scrapy_crawlera/middleware.py

```diff
@@ -211,3 +214,12 @@ def _is_crawlera_header(self, header_name):
             return False
         header_name = header_name.decode('utf-8').lower()
         return header_name.startswith(self.header_prefix.lower())
+
+    def _set_crawlera_default_headers(self, request):
+        has_crawlera_ua = 'X-Crawlera-UA' in request.headers
+        for header, value in self._headers:
+            if value is None:
+                continue
+            if has_crawlera_ua and header == 'X-Crawlera-Profile':
```

Comment:
Why does `X-Crawlera-UA` has precedence over `X-Crawlera-Profile`?

Reply:
If X-Crawlera-UA is set on the req, my understanding is that you want it rather than anything set on the DEFAULT_CRAWLERA_HEADERS.
According to https://doc.scrapinghub.com/crawlera.html#x-crawlera-profile `Crawlera won’t respect X-Crawlera-UA setting when X-Crawlera-Profile is set.`

Reply:
Ah, makes sense. :)

Reply:
But giving it a bit of thought, isn't setting `X-Crawlera-UA` via request and `X-Crawlera-Profile` via `DEFAULT_CRAWLERA_HEADERS` a developer's misusage of Crawlera?

I mean, should we fix it here in the middleware code, given that the API docs explicitly state that UA would be ignored when a profile is set?

I think that by cleaning up the `X-Crawlera-Profile` we're adding current Crawlera's semantics to the middleware.


Thoughts on this?

Reply:
So, the situation I am trying to avoid is:

Spider sets:
```
DEFAULT_CRAWLERA_HEADERS = {
  'X-Crawlera-Profile': 'desktop'
}
```

A request down the line sets `X-Crawlera-UA: desktop`, this will not work.

Is this bad usage or a bug? I feel like it's a bug of the middleware to not account for this, but I can see other points of view.

Reply:
I think it would be bad understanding of Crawlera headers semantics from the user. For me, it's as if the user would set this:

```python
DEFAULT_CRAWLERA_HEADERS = {
    'X-Crawlera-Profile': 'desktop',
    'X-Crawlera-UA': 'desktop',
}
```

I know that the example I gave is a bit too simplistic, specially considering that the mistakes you are concerned about are more prone to happen in big projects where the developers may not have a good view of all the settings/middlewares.

What about just logging a warning stating that current Crawlera implementation would ignore the `X-Crawlera-UA`? Or maybe a deprecation warning, given that this header is getting deprecated.

Not sure, but I'd prefer to avoid mixing Crawlera headers semantics in the middleware.

Just my opinion and would love to hear other people's opinions about this.

Reply:
I like the warning and deprecation warning more than my initial approach.

Not sure which one I would rather have to be honest. Will put some thought on it.

Reply:
:+1:  to the logging. Is a good way to gain attention from developer.